kuma/static/styles/includes/_mixins.scss

```diff
@@ -330,26 +332,108 @@ Spacing
 
 /*
 Bidi / l10n
+- The bidi mixins are used to declare both the LTR and RTL values for
+  a property together and output appropriate declarations
+- This makes it easier to maintain the RTL version since you only have
+  to update values in one place.
+- Each mixin has different uses and limitations
 ====================================================================== */
 
-// Allows setting of a property for LTR and RTL without having to deal with duplicating and maintaining selectors:
-// example: @include bidi-style(left, 20px, right, auto)
-@mixin bidi-style($ltr-prop, $value, $inverse-prop, $inverse-value, $make-important: false) {
-    $make-important: if($make-important, unquote('!important'), unquote(''));
-
-    #{$ltr-prop}: $value $make-important;
+/*
+    @mixin bidi
+    - Great for a declaration with multiple properties
+    - Does not support !important
+
+    A list with three properties overrides the ltr value in rtl locales
+    @include bidi(((float, left, right),))
+    LTR: float: left;
+    RTL: float: right;
+
+    A list with four properties overrides the ltr value in rtl locales
+    AND provides a new rtl only property value
+    @include bidi(((margin-right, 20px, margin-left, 0),))
+    LTR: margin-right: 10px;
+    RTL: margin-right: 0;
+         margin-left: 10px;
+
+    Example input:
+
+    .test {
+        @include bidi((
+            (float, left, right),
+            (margin-right, ($grid-spacing / 2), margin-left, 0),
+        ));
+    }
+
+    Example output:
+
+    .test {
+        float: left;
+        margin-right: 10px;
+    }
+    html[dir='rtl'] .test {
+        float: right;
+        margin-right: 0;
+        margin-left: 10px;
+    }
+}
+*/
+@mixin bidi($list) {
+    & {
+        @each $property, $ltr-value, $inverse-val-or-prop, $inverse-value in $list {
+            @if type-of($ltr-value) == 'null' or type-of($inverse-val-or-prop) == 'null' {
+                @warn $list; /* I don't know why this won't output as part of the error */
+                @error 'Error processing @mixin bidi(), see previous warning for stack trace';
+                /*
+                    Trouble shooting:
+                    - Check that you have the right number of nested (((parentheses)))
+                    - Check that there is a trailing comma
```

Comment:
@stephaniehobson Should there always be a trailing comma? Or only in these instances:

```
@include bidi(((float, left, right),));
```

Reply:
Yes, there should always be a trailing comma.

Technically, it works in some situations without it but we should be consistent.

Reply:
In case you're curious: it seems to work without the comma if there are no spaces or parentheses in the declaration.